package/kernel/gpio-nct5104d/src/gpio-nct5104d.c

```diff
@@ -270,45 +270,17 @@ static int nct5104d_gpio_probe(struct platform_device *pdev)
 	}
 	data->sio = sio;
 
-	platform_set_drvdata(pdev, data);
-
 	/* For each GPIO bank, register a GPIO chip. */
 	for (i = 0; i < data->nr_bank; i++) {
 		struct nct5104d_gpio_bank *bank = &data->bank[i];
 
 		bank->chip.parent = &pdev->dev;
 		bank->data = data;
 
-		err = gpiochip_add(&bank->chip);
-		if (err) {
-			dev_err(&pdev->dev,
-				"Failed to register gpiochip %d: %d\n",
-				i, err);
-			goto err_gpiochip;
-		}
-	}
-
-	return 0;
-
-err_gpiochip:
-	for (i = i - 1; i >= 0; i--) {
-		struct nct5104d_gpio_bank *bank = &data->bank[i];
-
-		gpiochip_remove (&bank->chip);
-	}
-
-	return err;
-}
-
-static int nct5104d_gpio_remove(struct platform_device *pdev)
-{
-	int i;
-	struct nct5104d_gpio_data *data = platform_get_drvdata(pdev);
-
-	for (i = 0; i < data->nr_bank; i++) {
-		struct nct5104d_gpio_bank *bank = &data->bank[i];
-
-		gpiochip_remove (&bank->chip);
+		err = devm_gpiochip_add_data(&pdev->dev, &bank->chip, bank);
```

Comment:
Why do you set data?
I think it should look like this:
```suggestion
		err = devm_gpiochip_add_data(&pdev->dev, &bank->chip, NULL);
```

Reply:
what's wrong with setting data?

Reply:
Changed to NULL.

Reply:
Ah nvm, I see the issue. gpiochip_get_data was missing.

Reply:
This was not wrong, but the old code did not set the data attribute and it is also not used.